### DISCUSSION TITLE
[FAQ] - DXC Auto-mode Setting / PHONE & CW Identification

### DISCUSSION THREAD

**Discussion by foldynl** [foldynl/QLog#279] [Q&A]
***Q***: Is it possible for QLog to switch modes when selecting a QSO from the DX Cluster?

***A***: Almost everything is possible; the question is whether it will be reliable enough. The reason I haven't implemented this yet is that the DX Spot does not contain information about the mode; it only provides the frequency. While it's true that you can deduce the mode from the frequency but it is only an estimation. Let me provide an example. If the frequency of a CW spot is very close to the FT8 segment, it may result in the CW being incorrectly identified as USB (FT8), causing QLog to switch modes inaccurately. Another issue arises when identifying the FT8 mode; some rigs need to switch to USB, while others require USB-Data mode. Another example is RTTY mode, which has its own mode on newer rigs. Determining this mode from the DX Cluster based on frequency is not deterministic. Although it is possible to extract this information from the Comment field, unfortunately, this field is free text and not well-structured for machine processing. This also applies to the identification of CW and PRONE spots in QLog, where identification is based on frequency. If a spot is in a segment for another type of operation, the identification may be incorrect.

Please feel free to engage in the discussion below. QLog is a community project, and it's not solely based on my opinion, which may be subject to errors. If we collectively find a method for accurately identifying modes, I would welcome it.

**Comment by mkimr** [comment]
My current experience of CQ-WW-CW is that many of the stations are spread far beyond the FT-8 segment. It could be argued that QLog is not a contest log, but a number of hams are taking advantage of this opportunity to acquire new DXCC countries. 
The lack of mode information DX spots is indeed a long-standing and sore point.

**Comment by foldynl** [comment]
Do you have any hint how to improve it in QLog? 

I don't want to train a neural network to recognize an unstructured comment field, and above all, it would be a waste of energy for this task.

I have already introduced frequency segments that determine at least CW/PHONE/FT8, but it seems that it is insufficient. Maybe set of comment field patterns would improved the situation. Maybe QLog should go in this direction.

**Reply by rsaue** [reply]
I am not happy with the way QLog does this today. If an deterministic approach , as Ladislav called it,  related to band plans will be right for  e.g. more than 90 % of the actual spots, that will be better than today.  This is mostly an CW issue, and when the band use "flow out during contests", there can be an user selected switch in QLog to expand the bandplan for any given period.

73 de LB2EG Richard

**Reply by foldynl** [reply]
I would like to understand you correctly. Are you not happy with the fact that when double-clicking on a spot and if the spot is marked as PHONE, QLog does not switch the rig to USB/LSB base on the band? Or are you not happy with how QLog now identifies Phone/CW or FT8?

**Comment by rsaue** [comment]
It is how the mode is communicated to the rig.  I do not do SSB, but when I click on a spot marked as CW my rig does not change into CW mode…

Edit: This was with my Flex-6300 on 2.  home shack . Will check with my other rigs tomorrow.

**Comment by rsaue** [comment]
Hi Ladislav, reading your initial and you subsequent postings again, I understand that mode change when clicking on a spot is not yet implemented i QLog software. Most of the time I use  the cluster for CW mode operation ,and filter spots for CW only. Looking at my Anan/Thetis TCP/IP  CAT Log  (Kenwood protocol) shows that mode and filter  are sent empty from QLog to the radio, and therefore read as querys only. I was not aware that mode change by spot was not implemented yet as my rig in such cases already was set in CW mode. Sorry for that.  

As a starting point spots should be filtered from the dx cluster upon band plans. Most of the operations on the band apply to band plan. As long as the comment field is not to be used for filtering purposes, the band plans is the only solution.  In some other logging software (e.g. UCXlog) the contest situation is handled by extending the CW band portion during contests. 

As for now, I suggest you add at least the mode change command in CAT command to the rig when clicked on a spot. 

73 de LB2EG Richard

**Comment by foldynl** [comment]
QLog already contains a band plan bit it is insufficient. I'm going to give a specific case where I think it is not possible to determine the exact mode. I may find that my thoughts are wrong.

DX Spot contains only following information
- spotter
- frequency
- DX call
- comment

therefore, only frequency can be used. QLog contains an internal bandplan. So, for example, for 20m it is this:
- freq >= 14.000 AND freq < 14.070) -> MODE_CW;
- (freq >= 14.070 AND freq < 14.074) -> MODE_DIGITAL;
- (freq >= 14.074 AND freq < 14.076) ->  MODE_FT8;
- (freq >= 14.076 AND freq < 14.098) -> MODE_DIGITAL;
- (freq >= 14.098 AND freq < 14.112) -> MODE_CW;
- (freq >= 14.112 AND freq < 14.350) -> MODE_USB;

OK. I know. There is already a problem here, because different regions have different bandplans. Let's try to forget about this problem and continue.

Let's start with a simple example of a spot that has a frequency of FT8.

1) It may be a CW spot that is close to the band. We are already solved this issue
2) However, if it really is a FT8 spot, then which mode should be sent to Rig? The answer can be simple: USB. But with the modern rigs, this is not true. Newer rigs use the so-called USB-Data mode that it has to be sent to rig

Another example could be RTTY. This type of spots is detected in the DIGI segment. The simple answer is to use USB but the modern rigs contain RTTY mode (or USB-Data) which should be used in this case.

if I summarize it, I see these problems

1) bandplan may differ from region to region
2) Flow out issue - CW spot in the DIGI (FT8) segment
3) Using specific modes for modern rigs

This led me to the fact that rather than sending an incorrect mode to the rig, I'd rather not send mode and leave it to the operator to choose a correct mode.

However, if the general opinion is that it is better to switch incorrectly than do not switch, then I can implement the automatic mode switching in QLog.

**Comment by rsaue** [comment]
Hi,
You could make this optional, with a standard bandplan that could be user editable. Then everybody would be happy. 

This is done in LogHX, as shown in the enclosed screendump from my setup. 

73 de LB2EG Richard

 
![bandplan loghx3](https://github.com/foldynl/QLog/assets/22942440/86ff76fa-4190-45f7-a517-f98e8fcdff55)

**Reply by foldynl** [reply]
Richard, it is not about bandplan, because qlog already contains a bandplan. It's about which mode to send to the rig. 

I'll try to design something and we will see if it will work.

**Reply by rsaue** [reply]
I understand, but in the example I enclosed i my prior post the mode is also (user editable) allocated to the band plan or portions of it.

I am looking forward to your design. 

73 de LB2EG Richard
